spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-sample/src/main/java/com/example/PubSubApplication.java

```diff
@@ -20,7 +20,7 @@
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
-public class PubSubApplication {
+public abstract class PubSubApplication {
```

Comment:
That's interesting. We pursue the `HideUtilityClassConstructorCheck` rule, but SonarQube doesn't like our fix and suggests its own goodness 😄 
However, at the same, time you have already excluded samples for that `HideUtilityClassConstructorCheck` rule, so maybe it's time to remove all those `abstract` modifications?..

Reply:
JDK seems to like private constructors for utility method classes. I would go with that rather than the `abstract` trick. That would satisfy Sonar and Spring Style. :smile:

Reply:
Huh? This is sample application. I wouldn't overcomplicate end-user experience with extra code from us.

Reply:
I was actually thinking of `ConversionUtils`. The samples should just be excluded indeed.